vterm.el

```diff
@@ -432,14 +447,34 @@ This is the value of `next-error-function' in Compilation buffers."
     (use-local-map vterm-mode-map)
     (vterm-send-start)))
 
-(defun vterm-copy-mode-done ()
-  "Save the active region to the kill ring and exit `vterm-copy-mode'."
-  (interactive)
+(defun vterm-copy-mode-done (arg)
+  "If a region is active then kill it, otherwise kill the current line, then exit `vterm-copy-mode'.
+
+If a region is defined then that region is killed, with no region then
+current line is killed from start to end.
+
+The option `vterm-copy-exclude-prompt` controls if the prompt
+should be included in a line copy.  Using the universal prefix
+will invert `vterm-copy-exclude-prompt` for that call."
+  (interactive "P")
   (unless vterm-copy-mode
     (user-error "This command is effective only in vterm-copy-mode"))
-  (unless (region-active-p)
-    (user-error "No region is active"))
-  (kill-ring-save (region-beginning) (region-end))
+  (save-excursion
+    (unless (region-active-p)
+      (beginning-of-line)
+      ;; Are we excluding the prompt?
+      (when (or (and vterm-copy-exclude-prompt (not arg))
+                (and (not vterm-copy-exclude-prompt) arg))
+        ;; Using vterm's prompt detection
+        (if vterm-copy-use-vterm-prompt
+            (vterm-beginning-of-line)
+          ;; Using vterm-copy-prompt-regexp
+          (unless (re-search-forward vterm-copy-prompt-regexp (save-excursion (end-of-line) (point)) 'noerror)
+            ;; Move to the front of the line if no prompt found
+            (beginning-of-line))))
+      (set-mark (point))
+      (end-of-line))
```

Comment:
Can you skip the `fake-newline` 
You can check whether the newline is faked by `(get-text-property (point) 'vterm-line-wrap)` when the point at the end of line .
and see `vterm--filter-buffer-substring` and `vterm--remove-fake-newlines`  and https://github.com/akermu/emacs-libvterm/pull/274

Reply:
That is a very good idea.  I'll have to read about `fake-newlines` and see how to do that.